tests/test_store/test_memory.py

```diff
@@ -46,6 +52,28 @@ def test_store_supports_partial_writes(self, store: MemoryStore) -> None:
     def test_list_prefix(self, store: MemoryStore) -> None:
         assert True
 
+    @pytest.mark.parametrize("dtype", ["uint8", "float32", "int64"])
+    @pytest.mark.parametrize("zarr_format", [2, 3])
+    async def test_deterministic_size(
+        self, store: MemoryStore, dtype, zarr_format: ZarrFormat
+    ) -> None:
+        def padding_size() -> int:
+            a = zarr.empty(
+                store=store,
+                shape=(3,),
+                chunks=(1000,),
+                dtype=dtype,
+                zarr_format=zarr_format,
+                overwrite=True,
+            )
+            a[...] = 1
+            key = "0" if zarr_format == 2 else "c/0"
+            return len(store._store_dict[key])
+
+        l1 = padding_size()
+        l2 = padding_size()
+        assert l1 == l2
```

Comment:
It would be good to add another check here to compare l1 and l2 to the exact value expected (1000?)